russstation/tools/spacemerge/spacemerge.py

```diff
@@ -1,597 +0,0 @@
-def get_honk_files(repo):
-	# -I ignore binary, -i case insenstive, -l filename instead of matched lines, -E extended regex, -e pattern is next parameter
-	honk_paths = set(repo.git.grep("-IilEe", honk_check_pattern).splitlines())
-	printv("Retrieved list of files with honk comments")
-	return honk_paths
-
-# get conflicted files, process em, revert unchanged conflicts
-def process_conflicts(repo, upstream_paths):
-	print("Processing files...")
-	# U for unmerged ie conflicted
-	conflict_paths = set(repo.git.diff("--name-only", "--diff-filter=U").splitlines())
-	printv(len(conflict_paths), "unmerged files")
-	honk_paths = get_honk_files(repo)
-	# the paths we will process are unmerged, have honk comments, and were changed upstream
-	process_paths = conflict_paths.intersection(upstream_paths).intersection(honk_paths)
-	printv(len(process_paths), "files to process")
-	for path in process_paths:
-		try:
-			resolve_conflicts(repo, path)
-		except:
-			printv("Failed processing on", path)
-			printv("Exception:", sys.exc_info())
-	# paths that weren't changed upstream need reverted
-	print("Reverting unchanged files...")
-	revert_paths = conflict_paths - upstream_paths
-	printv(len(revert_paths), "unchanged files to revert")
-	for path in revert_paths:
-		try:
-			# why doesn't git have a single operation for unstage and discard
-			printvv("Reverting", repo.git.status("-s", path))
-			repo.git.restore(path, "--staged")
-			repo.git.restore(path)
-		except git.exc.GitCommandError as e:
-			# git restore fails if file doesn't exist because it was git removed
-			printv("Couldn't revert", path)
-			printv(repr(e))
-	# remaining files just checkout theirs
-	print("Resolving remaining conflicts...")
-	their_paths = conflict_paths.intersection(upstream_paths) - honk_paths
-	printv(len(their_paths), "remaining conflicted files")
-	for path in their_paths:
-		try:
-			repo.git.checkout("--theirs", path)
-			repo.git.add(path)
-			printvv("Checked out their", path)
-		except:
-			printv("Failed to checkout their", path)
-	# any remaining modified files don't have conflicts, but may have honks
-	# apparently Unmerged is a subset of Modified, so to add all the remaining files we need more set operations
-	print("Resolving remaining nonconflicts...")
-	modified_paths = set(repo.git.diff("--name-only", "--diff-filter=M").splitlines())
-	true_modified_paths = modified_paths - conflict_paths
-	modified_honk_paths = true_modified_paths.intersection(honk_paths)
-	modified_their_paths = true_modified_paths - honk_paths
-	for path in modified_honk_paths:
-		# don't keep conflict sections during resolution
-		resolve_conflicts(repo, path, keep_conflicts = False)
-	for path in modified_their_paths:
-		repo.git.checkout("--theirs", path)
-		repo.git.add(path)
-		printvv("Checked out their", path)
-
-# find conflict section, inspect section based on rules, modify file, stage in git
-def resolve_conflicts(repo, filename, keep_conflicts = True):
```

Comment:
so, you can't just toss spacemerge. even if we unify the `.dme`, sometimes we have changes in tg files that need to not get overwritten on automerges. nothing in this PR handles these cases, which we need this function for. otherwise merging is an absolute pain because of how much manual review is required